Scripts/ElfStats.gd

```diff
@@ -90,14 +90,20 @@ static func add_item(item) -> void:
 	ElfStats._items[item.name] = item
 	
 	for s in ElfStats._stats:
-		for i in range(0, s.modifiers.size(), 1):
-			if s.modifiers[i].item_name == item.name:
-				s.remove_modifier(s.modifiers[i])
-				break
-		 
-		for c in item.stat_modifiers:
-			if s.is_named(c.stat_name):
-				s.add_modifier(c)
+		_remove_old_item_modifier(s, item)
+		
+		_add_new_item_modifier(s, item)
+				
+static func _remove_old_item_modifier(stat: Resource, item: Resource) -> void:
```

Comment:
Chciałem żeby zmienna `stat` w tej funkcji była referencją do obiektu z `ElfStats._stats`, jednak według [tego linku](https://godotengine.org/qa/32684/how-do-i-pass-variables-by-reference-to-functions) :

> In gdscript you do not have control over this. All simple types go as values, all complex types go as reference.

Nie jestem pewien czy mój sposób zadziała w przypadku klasy Stat 🤔 
Co o tym myślisz @dyedmagenta ?

Reply:
Chyba nie dokońca rozumiem co chcesz zrobić?
Resource nie jest typem prostym więc powinien być referencją według twojego linku.
Jeżeli chcesz tylko odwoływać się do zmiennej _stats w ElfStats to parametr `stat: Resource` jest niepotrzebny.

Jeśli chodzi tobie o to żeby definicja funkcji wyglądała tak:

```
static func _remove_old_item_modifier(stat: Stat, item: Resource) -> void:
 	.
	.
	.
```

To to się nazywa statyczne typowanie. [Tutaj](https://docs.godotengine.org/en/stable/getting_started/scripting/gdscript/static_typing.html#static-typing-in-gdscript) jest to wytłumaczone w dokumentacji.
TLDR: Wystarczy że dodasz do Stat.gd linijkę `class_name Stat`